optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -741,11 +742,11 @@ public boolean hasEntityDescriptor(Class<?> entitySubclass) {
             // Reverse order to find the nearest ancestor
             for (Class<?> entityClass : reversedEntityClassList) {
                 if (entityClass.isAssignableFrom(entitySubclass)) {
-                    return entityDescriptorMap.get(entityClass);
+                    return Optional.of(entityDescriptorMap.get(entityClass));
                 }
             }
-            return null;
-        });
+            return Optional.empty();
```

Comment:
Insert a log statement before this line and try with the Nurse Rostering example to see the effect.